source/input_types.rst

```diff
@@ -0,0 +1,192 @@
+
+
+.. _csv:
+
+CSV
+---
+A table of data stored in a .csv (comma-separated value) format. You may use any spreadsheet software to create and edit a table, and then export it to CSV format.
+
+
+.. _raster:
+
+raster
+------
+A spatial data file that stores data in pixels, which each have a numeric value.
+InVEST accepts all GDAL-supported raster formats, including GeoTiff (.tif) and many more.
+For the complete list, see https://gdal.org/drivers/raster/index.html.
+
+Bands
+~~~~~
+Rasters may have multiple bands. All InVEST models look at the first band only. If you are using a multi-band raster,
+please be sure that the correct dataset is in the first band.
+
+
+.. _datatypes:
+
+Data types
+~~~~~~~~~~
+Every raster has a *data type* which determines the minimum and maximum value that each pixel can have. Some data types allow positive and negative numbers, while others only allow positive numbers. Most of the time you will not need to change your raster's data type, but it is important to be aware of.
+
+Understanding data types
+^^^^^^^^^^^^^^^^^^^^^^^^
+A raster's data type is displayed as a code in the raster metadata, which may be viewed in your GIS software. A data type code has 3 components:
+
+1. Unsigned marker (**u**), optional
+
+   By default, data types are signed, meaning that they include a sign (+ or -) indicating whether the number is positive or negative.
+   You need a signed data type to store negative data.
+   If a data type begins with a **u**, that means it is unsigned. All unsigned data is positive. If you do not need to store negative data,myou can save space by using an unsigned type.
+
+   This distinction only exists for integer data types. Float types are always signed.
+
+2. Type (**float** or **int**)
+
+   Floating-point (float) types can store digits after the decimal point. There is no hard limit on how many decimal places they can store, but they are only accurate to a limited number of decimal places.
+   Integer (int) types can only store whole numbers. They have perfect accuracy.
+   It is best to use integer data types when possible for discrete data.
+
+3. Size (**8, 16, 32, 64**)
+
+   This is how many bits are used to store the number. It determines the range of numbers that can fit into the data type.
+   You can save space by using the smallest size that works for your data. For example, the large numbers available in a **float64** raster are rarely needed. The **float32** range is sufficient for most real-world data, and it uses half as much space.
+   The **uint8** type is sufficient for most discrete data that InVEST uses (land use/land cover classes, soil groups, and so on) which have fewer than 256 possible values.
+
+Here are all the standard raster data types and their ranges (ranges include the starting and ending values):
+
+- **byte** (**uint8**): any integer from 0 to 255
```

Comment:
Arc users in particular are a source of signed byte rasters, with values from -128 to 127, and which GDAL does not always handle in a way that is expected since GDAL does not have a formal definition of an int8 type.

I _think_ my recommendation here is to advise against using signed integer rasters in favor of unsigned byte rasters, but maybe we've fixed all those signed byte issues and my memory is out of date.  Thoughts?

Reply:
I would love to know the history of why GDAL uses that `PIXELTYPE=SIGNEDBYTE` attribute instead of just having an int8 type! Seems so counterintuitive. I know `new_raster_from_base` _does_ handle signed bytes, and that still tripped me up in the Stormwater model. We do not normally use signed byte rasters in our tests so I'm sure there's some problems lurking out there.
I'll add a note advising against signed byte rasters. If we ever get around to natcap/invest#536, testing everything with signed byte rasters would be a good use for it.

Reply:
>I would love to know the history of why GDAL uses that PIXELTYPE=SIGNEDBYTE attribute instead of just having an int8 type! 

Likewise!  The GDAL commit history on github is unfortunately not very helpful, so my best guess is that it was added somewhere in the CVS history, before GDAL development moved to SVN and then eventually git.

> I'll add a note advising against signed byte rasters. If we ever get around to natcap/invest#536, testing everything with signed byte rasters would be a good use for it.

That sounds good to me!  And yes, I agree about testing against signed byte rasters if/when we get to that extra testing would be a really good use for it.